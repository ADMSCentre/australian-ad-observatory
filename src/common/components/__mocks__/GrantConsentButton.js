/* eslint-disable react/prop-types */
import React from 'react';

const GrantConsentButton = ({ label }) => <button>{label}</button>;

export default GrantConsentButton;
