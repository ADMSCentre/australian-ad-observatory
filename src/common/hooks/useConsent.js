import { useCallback, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import actions from 'common/state/consent/actions';

const useConsent = () => {
  const { granted, acceptedAt, viewedAt } = useSelector(
    ({ consent: { granted, acceptedAt, viewedAt } }) => ({
      granted,
      acceptedAt,
      viewedAt
    })
  );

  const dispatch = useDispatch();

  const loadConsent = useCallback(() => dispatch(actions.loadConsent()), [dispatch]);

  const saveConsent = useCallback(() => dispatch(actions.saveConsent()), [dispatch]);

  const viewedConsent = useCallback(() => dispatch(actions.viewedConsent()), [dispatch]);

  useEffect(() => {
    /* istanbul ignore else */
    if (!granted) loadConsent();
  }, [granted]);

  return { granted, acceptedAt, viewedAt, saveConsent, viewedConsent };
};

export default useConsent;
