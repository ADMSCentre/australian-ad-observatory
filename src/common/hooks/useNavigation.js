import { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import actions from 'common/state/navigation/actions';

const useNavigation = () => {
  const current = useSelector(({ navigation: { current } }) => current);

  const dispatch = useDispatch();

  const goto = useCallback(destination => dispatch(actions.navigate(destination)), [dispatch]);

  return { current, goto };
};

export default useNavigation;
