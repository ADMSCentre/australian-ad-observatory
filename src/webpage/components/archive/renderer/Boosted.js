import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faRocket } from '@fortawesome/free-solid-svg-icons';

const Boosted = ({ boosted }) =>
  boosted ? (
    <div className="btn btn-outline-success">
      <FontAwesomeIcon icon={faRocket} />
      Post was boosted
    </div>
  ) : null;

Boosted.displayName = 'Boosted';

Boosted.propTypes = { boosted: PropTypes.bool };

Boosted.defaultProps = { boosted: false };

export default Boosted;
