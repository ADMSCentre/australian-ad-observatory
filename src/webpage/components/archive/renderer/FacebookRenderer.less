.fb-item {
  padding: 0;
  width: 500px;
  background-color: #fff;
  border-radius: 5px;
  border: 1px solid #ddd;
  margin-top: 25px;
  margin-bottom: 15px;
  margin-right: 15px;

  &:first-child {
    padding: 12px 12px 0 12px;
  }

  .hidden_elem {
    display: none;
  }

  [data-testid="post_message"] {
    padding: 10px 0;
  }

  & > img {
    max-width: 90%;
    border: 0;
    margin: 0;
    padding: 0;
  }

  video {
    display: none;
  }

  div.mtm div div div span div img {
    width: 498px;
    height: auto;
    margin-left: -12px;
    border-top: 1px solid #ccc;
    border-bottom: 1px solid #ccc;
  }

  // Hide divs that have been otherwise rendering as stray periods.
  ._6spk {
    display: none;
  }

  // Don't allow user to click buttons or links.
  // a,
  // button {
  //   // pointer-events: none;
  // }

  h5 {
    font-size: 14px; // Match size of source bar w/ Facebook's size
    margin-bottom: 0; // Bring timestamp closer to the source bar title
  }

  // Add "arrow" in place of "to" in source bar.
  ._gb8.img::after {
    content: ">";
    font-style: normal;
  }

  // Remove word "to" in source bar.
  ._gb8 u {
    display: none;
  }

  ._6m6 {
    padding-top: 2px;
    margin-top: 3px;
    display: block;
    float: left;
    padding-bottom: 10px;

    a {
      color: #1d2129;
      font-size: 16px;
      font-weight: 600;

      &:hover {
        text-decoration: none;
        color: #1d2129;
      }
    }
  }

  // Bring author names inline with profile pics.
  ._6a._6b {
    display: inline-block;
    height: 20px !important;

    a {
      color: #385898;
      font-size: 14px;
      font-weight: 600;
      font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
      -webkit-font-smoothing: subpixel-antialiased;
    }
  }

  ._ohe ._6a {
    display: block;
    float: right;
    padding-left: 18px;
    height: 20px;
    vertical-align: middle;

    a {
      border: 1px solid rgb(190, 195, 201);
      background-color: rgb(238, 240, 242);
      color: #4b4f56;
      padding: 5px 8px;
      border-radius: 2px;

      &:hover {
        text-decoration: none;
      }
    }
  }

  ._6m7 {
    margin-top: 3px;
    color: #606770;
    font-size: 14px;
    line-height: 20px;
    word-break: break-word;
    display: none;
    overflow-x: hidden;
  }

  div[data-testid="post_message"] p {
    color: #1d2129;
    font-size: 14px;
    font-weight: 400;
    margin-bottom: 5px;
  }

  .accessible_elem {
    display: none;
  }

  ._4r_y {
    height: 2px;
  }

  // Hide inconsequential artifacts.
  ._52c6 {
    display: none;
  }

  div ._3x-2 {
    margin: 0;
    padding: 0;
  }

  // Hide buttons.
  button {
    display: none;
  }

  // Profile pic wrapper.
  ._5pb8._1yz2._8o._8s.lfloat._ohe,
  ._38vo {
    display: block;
    float: left;
    width: 32px;
    margin-right: 8px;
    border-radius: 16px;
    overflow: hidden;
  }

  ._38vo {
    max-height: 32px;
  }

  ._38vo img {
    max-height: 32px;
  }

  ._5x46,
  ._3dp ._29k {
    /* Profile name wrapper */
    margin-bottom: 6px;
  }

  ._3ekx {
    display: block;
  }

  ._6lz {
    padding-top: 10px;
    color: #606770;
    font-size: 12px;
  }

  .uiScaledImageContainer {
    max-width: 100%;
    width: auto;
    height: auto;
  }

  .scaledImageFitWidth {
    // max-width: 100%;
    // width: auto;
    // height: auto;
  }

  .ellipsis {
    line-height: 14px;
    text-transform: uppercase;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  .card-title {
    display: none;
  }

  .card-subtitle {
    display: none;
  }

  &.fb-publicUserPost {
    // background-color: #e1edec;
    div.mtm div div div span div div {
      display: none;
    }
  }

  &.fb-publicPagePost {
    // background-color: #e1edec;
    div.mtm div div div span div div {
      display: none;
    }
  }

  &.fb-sponsoredPost {
    // background-color: #f1edec;

    a[role="button"] {
      //border: 1px solid;
      border-radius: 2px;
      box-sizing: content-box;
      font-size: 12px;
      -webkit-font-smoothing: antialiased;
      font-weight: bold;
      justify-content: center;
      padding: 0 8px;
      position: relative;
      text-align: center;
      text-shadow: none;
      vertical-align: middle;
    }
  }

  ul.uiList {
    margin-left: -52px;
  }

  ul.uiList li {
    display: none;
  }

  ul.uiList li:first-child {
    display: inline-block;
    border-top: 1px solid #ccc;
    width: 100%;
  }

  ul.uiList li img {
    width: 498px;
    height: auto;
  }

  ._5eut,
  ._42gt {
    display: none;
  }

  ._4lpe,
  ._3htz,
  ._5mly,
  ._40ja,
  ._45oh {
    display: none;
  }
}

.targeting {
  margin-left: 5px;
  width: 500px;

  h4 {
    font-size: 12px;
    font-weight: 600;
    margin: 0;
    padding: 0;
  }

  .reason {
    margin-right: 10px;

    .reason-name {
      padding-left: 5px;
    }
  }
}
