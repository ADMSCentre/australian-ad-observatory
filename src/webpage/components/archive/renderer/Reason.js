import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faUsers,
  faGlobe,
  faTransgenderAlt,
  faMapMarker,
  faDotCircle
} from '@fortawesome/free-solid-svg-icons';

const REASONS = {
  WAISTUICustomAudienceType: {
    name: 'Custom Audience',
    icon: faUsers
  },
  WAISTUILocaleType: {
    name: 'Locale',
    icon: faGlobe
  },
  WAISTUIAgeGenderType: {
    name: 'Age and Gender',
    icon: faTransgenderAlt
  },
  WAISTUILocationType: {
    name: 'Location',
    icon: faMapMarker
  }
};

const Reason = ({ reason }) =>
  REASONS[reason] ? (
    <div className="reason badge badge-primary">
      <FontAwesomeIcon icon={REASONS[reason].icon} />
      <span className="reason-name">{REASONS[reason].name}</span>
    </div>
  ) : (
    <div className="reason badge badge-primary">
      <FontAwesomeIcon icon={faDotCircle} />
      <span className="reason-name">{reason}</span>
    </div>
  );

Reason.displayName = 'Reason';

Reason.propTypes = {
  reason: PropTypes.string.isRequired
};

export default Reason;
