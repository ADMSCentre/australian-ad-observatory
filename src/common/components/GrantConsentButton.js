import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

import useConsent from 'common/hooks/useConsent';
import useNavigation from 'common/hooks/useNavigation';

const GrantConsentButton = ({ label, to }) => {
  const { granted, saveConsent } = useConsent();
  const { goto } = useNavigation();

  const saveAndGo = () => {
    saveConsent();
    if (to) goto(to);
  };

  return granted === null || granted === 'current' ? null : (
    <Button variant="primary" onClick={saveAndGo} label={label} />
  );
};
GrantConsentButton.displayName = 'Grant aware button';

GrantConsentButton.propTypes = {
  label: PropTypes.string.isRequired,
  to: PropTypes.string
};

GrantConsentButton.defaultProps = {
  to: null
};

export default GrantConsentButton;
