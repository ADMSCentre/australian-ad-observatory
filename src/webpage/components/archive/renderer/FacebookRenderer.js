import React from 'react';
import PropTypes from 'prop-types';

import Reason from './Reason';
import Boosted from './Boosted';

import './FacebookRenderer.less';

const itemShape = {
  id: PropTypes.string.isRequired,
  itemId: PropTypes.string.isRequired,
  itemType: PropTypes.string.isRequired,
  payload: PropTypes.shape({
    contentHtml: PropTypes.string.isRequired
  }).isRequired
};

const AdTargeting = ({ item }) => {
  if (!item.payload.adTargetingData) return null;

  const {
    data: { waist_targeting_data: targetingData, waist_is_marketplace_boosted_listing: boosted }
  } = item.payload.adTargetingData;
  const targetingReasons = targetingData.map(({ __typename }) => __typename);

  return (
    <div className="targeting">
      <h4>Ad Targeting</h4>
      <Boosted boosted={boosted} />
      {targetingReasons.map(reason => (
        <Reason key={reason} reason={reason} />
      ))}
    </div>
  );
};
AdTargeting.displayName = 'Ad targeting';

AdTargeting.propTypes = {
  item: PropTypes.shape(itemShape).isRequired
};

const FacebookRenderer = ({ item }) => {
  const {
    id,
    itemId: _itemId,
    itemType,
    payload: { contentHtml: html }
  } = item;

  const classes = `ati-item fb-item fb-${itemType}`;

  return (
    <div className="archive-item">
      <div id={id} className={classes} dangerouslySetInnerHTML={{ __html: html }} />
      <AdTargeting item={item} />
    </div>
  );
};

FacebookRenderer.propTypes = {
  item: PropTypes.shape(itemShape).isRequired
};

export default FacebookRenderer;
