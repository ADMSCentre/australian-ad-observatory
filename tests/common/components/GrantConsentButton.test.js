import React from 'react';
import renderer from 'react-test-renderer';
import useConsent from 'common/hooks/useConsent';
import useNavigation from 'common/hooks/useNavigation';
import { mount } from 'enzyme';

import GrantConsentButton from 'common/components/GrantConsentButton';

jest.mock('common/hooks/useConsent');
jest.mock('common/hooks/useNavigation');

const label = 'some label';
const to = 'some-tab';

const saveConsent = jest.fn();
const goto = jest.fn();

const cleanup = () => {
  useConsent.mockClear();
  saveConsent.mockClear();
  goto.mockClear();
};

const doTest = granted => {
  describe(`when granted is '${granted || 'null'}'`, () => {
    describe('check rendering', () => {
      let tree;

      beforeAll(() => {
        useConsent.mockReturnValue({ granted, saveConsent });
        useNavigation.mockReturnValue({ goto });
      });

      afterAll(cleanup);

      describe('when there is no to prop supplied', () => {
        beforeAll(() => {
          tree = renderer.create(<GrantConsentButton label={label} />);
        });

        it('renders correctly', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('when there is a to prop', () => {
        beforeAll(() => {
          tree = renderer.create(<GrantConsentButton label={label} to={to} />);
        });

        afterAll(cleanup);

        it('renders correctly', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });
  });
};

[null, 'none', 'old', 'current'].forEach(doTest);

describe('check action', () => {
  beforeAll(() => {
    useConsent.mockReturnValue({ granted: 'none', saveConsent });
    useNavigation.mockReturnValue({ goto });
  });

  describe('when there is a to prop', () => {
    beforeAll(() => {
      const wrapper = mount(<GrantConsentButton label={label} to={to} />);
      wrapper.find('button').simulate('click');
    });

    afterAll(cleanup);

    it('called saveConsent', () => {
      expect(saveConsent).toHaveBeenCalled();
    });

    it('called goto with the to value', () => {
      expect(goto).toHaveBeenCalledWith(to);
    });
  });

  describe('when there is not a to prop', () => {
    beforeAll(() => {
      const wrapper = mount(<GrantConsentButton label={label} />);
      wrapper.find('button').simulate('click');
    });

    afterAll(cleanup);

    it('called saveConsent', () => {
      expect(saveConsent).toHaveBeenCalled();
    });

    it('did not call goto', () => {
      expect(goto).not.toHaveBeenCalled();
    });
  });
});
