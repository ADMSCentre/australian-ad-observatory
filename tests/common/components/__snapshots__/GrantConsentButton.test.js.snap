// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when granted is 'current' check rendering when there is a to prop renders correctly 1`] = `null`;

exports[`when granted is 'current' check rendering when there is no to prop supplied renders correctly 1`] = `null`;

exports[`when granted is 'none' check rendering when there is a to prop renders correctly 1`] = `
<button
  className="btn btn-primary btn-block"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  some label
</button>
`;

exports[`when granted is 'none' check rendering when there is no to prop supplied renders correctly 1`] = `
<button
  className="btn btn-primary btn-block"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  some label
</button>
`;

exports[`when granted is 'null' check rendering when there is a to prop renders correctly 1`] = `null`;

exports[`when granted is 'null' check rendering when there is no to prop supplied renders correctly 1`] = `null`;

exports[`when granted is 'old' check rendering when there is a to prop renders correctly 1`] = `
<button
  className="btn btn-primary btn-block"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  some label
</button>
`;

exports[`when granted is 'old' check rendering when there is no to prop supplied renders correctly 1`] = `
<button
  className="btn btn-primary btn-block"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  some label
</button>
`;
